README.md

```diff
@@ -0,0 +1,19 @@
+# Zirgen Circuit Compiler
+
+Zirgen is a compiler for a domain-specific language, also called "zirgen",
+which creates arithmetic circuits for the RISC Zero proof system.
+
+For the most part, users of RISC Zero should be writing their zkApps in Rust
+and using our RISC-V zkVM rather than building them directly in Zirgen.
+Sometimes, however, it is necessary or desirable to write parts of an
+application as an arithmetic circuit to integrate directly with our proof system
+and achieve better performance. In fact, our zkVM itself is a circuit written
```

Comment:
Can we make it more explicit in the readme where to find the source code for the zkVM, and how the contents of this repo relate to what's in the public repo?

Reply:
(i'm already fielding questions about how the zkr/zir files relate to the poly_ext file)